docs/guides/implementation-guides/how-to-implement-dependency-injection.md

```diff
@@ -3,13 +3,142 @@ id: how-to-implement-dependency-injection
 title: How To Implement Dependency Injection
 ---
 
-import MvvmArchitectureDiagram from '/img/guides/implementation-guides/mvvm-architecture.png';
+# How To Implement Dependency Injection in Avalonia
 
-# How To Implement Dependency Injection
-
-Content in preparation.
+[Dependency injection (DI)](https://en.wikipedia.org/wiki/Dependency_injection) allows developers to write cleaner, more modular, and testable code. It accomplishes this by creating discrete services that are passed around/created as needed.
 
 This guide will show you how to use Dependency Injection (DI) with _Avalonia UI_ and the MVVM pattern. 
 
-<img src={MvvmArchitectureDiagram} alt=""/>
+Let's assume that you have an app with a MainViewModel, a BusinessService and a Repository. MainViewModel has a dependency to BusinessService and BusinessService to Repository. A simple implementation would look like this:
+
+```csharp
+public partial class MainViewModel : ObservableObject
+{
+    private readonly IBusinessService _businessService;
+
+    public MainViewModel(IBusinessService businessService)
+    {
+        _businessService = businessService;
+    }
+}
+```
+
+```csharp
+public class BusinessService : IBusinessService
+{
+    private readonly IRepository _repository;
+
+    public MainViewModel(IRepository repository)
+    {
+        _repository = repository;
+    }
+}
+```
+
+```csharp
+public class Repository : IRepository
+{
+}
+```
+
+Typically you would directly instantiate `Repository` and pass it into `BusinessService` then pass it into `MainViewModel`, like this:
+
+```csharp
+var window = new MainWindow
+{
+    DataContext = new MainViewModel(new BusinessService(new Repository()))
+}
+```
+
+ This works great for simple constructors that are not used very often and don't change. But this technique does not scale very well because:
+- The more dependencies your constructor has the more things you will need to instantiate and pass in. Instantiating the dependencies locally (such as by doing new MainViewModel(new MyService())) results in direct rigid coupling to a specific instance of the dependencies.
+- Similarly if MainViewModel creates its dependencies itself, (such as in the constructor body) it also becomes directly coupled to the creation of the dependencies which can result in mostly the same problems. 
+- Furthermore, if the object is instantiated in many places, every single reference to any of the dependencies would also need to be updated should the dependencies of `MainViewModel` ever change (such as by requiring additional dependencies or require a different implementation of a dependency). 
+
+Dependency injection solves these problem by abstracting away the creation of objects and their dependencies. This allows for well encapsulated services to be used that will be automatically passed into any other service that is registered to use them. 
+
+See below the steps you need to do in order to resolve that dependency using DI.
```

Comment:
I think it flows with the previous one which is listed the problems. I'm happy to removed it if you think it's needed

Reply:
The thing is you have already implied the steps by giving the step number. It's kind of like a chapter in a book directly saying the next chapter is about to start. It would be better to put this closer to the introduction if you wanted to keep it.